java/src/main/java/Client.java

```diff
@@ -0,0 +1,66 @@
+
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+public class Client {
+
+    public void start() {
+
+        var credential = createCredential();
+
+        var processor = new ServiceBusClientBuilder()
+                .fullyQualifiedNamespace(Settings.getNameSpace())
+                .credential(credential)
+                .processor()
+                .queueName(Settings.getQueueName())
+                .processMessage(this::processMessage)
+                .processError(this::processError)
+                .buildProcessorClient();
+
+        processor.start();
+
+    }
+
+    private ClientSecretCredential createCredential() {
+        return new ClientSecretCredentialBuilder()
+                .tenantId(Settings.getTenantId())
+                .clientId(Settings.getClientId())
+                .clientSecret(Settings.getSecret())
+                .build();
+    }
+
+    private void processMessage(ServiceBusReceivedMessageContext context) {
+        writeMessage(context.getMessage().getBody().toString());
+    }
+
+    private void processError(ServiceBusErrorContext context) {
+        writeMessage(context.getException().getMessage());
+    }
+
+    private void writeMessage(String message) {
+        try {
+            System.out.println("writing message in data file");
+            var path = getDownloadFilePath();
+            Files.writeString(path, message, StandardOpenOption.APPEND);
+        } catch (Exception ex) {
+            System.err.println(ex.getMessage());
+        }
+    }
+
+    private Path getDownloadFilePath() throws IOException {
+        var path = Path.of(Settings.getDownloadFilePath());
+        if (!Files.exists(path)) {
+            Files.createFile(path);
+        }
+        return path;
+    }
```

Comment:
This looks like it continuously keeps writing to `data.txt`, is that correct?

The pattern used in the other example clients has been to pull the dataset out of the message subject. As in [the docs](https://learn.microsoft.com/en-us/java/api/overview/azure/messaging-servicebus-readme?view=azure-java-stable#receive-messages), you should be able to pull the subject out of the ServiceBusReceivedMessage and then save a new file for each new message, rather than using one very large file, and name each file based off of the subject and time received.

Reply:
Sure, done